src/topology/homotopy/product.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2021 Praneeth Kolichala. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Praneeth Kolichala
+-/
+import topology.homotopy.basic
+import topology.constructions
+
+/-!
+# Product of homotopies
+
+In this file, we introduce definitions for the product of
+homotopies. We show that the products of relative homotopies
+are still relative homotopies.
+
+## Definitions
+- `homotopy.pi homotopies`: Let f and g be a family of functions
+  indexed on I, such that for each i ∈ I, fᵢ and gᵢ are maps from A to Xᵢ.
+  Let `homotopies` be a family of homotopies from fᵢ to gᵢ for each i.
+  Then `homotopy.pi homotopies` is the canonical homotopy
+  from ∏ f to ∏ g, where ∏ f is the product map from A to Πi, Xᵢ,
+  and similarly for ∏ g. All homotopies are done relative to some set S ⊆ A.
+
+- `homotopy.prod F G` is the product of homotopies F and G,
+   where F is a homotopy between f₀ and f₁, G is a homotopy between g₀ and g₁.
+   The result F × G is a homotopy between (f₀ × g₀) and (f₁ × g₁).
+   Again, all homotopies are done relative to S.
+-/
+
+noncomputable theory
+
+namespace continuous_map.homotopy
```

Comment:
(see comments below first)

The ones about `homotopy` should go in `continuous_map.homotopy`, and the ones about `homotopy_rel` should go in `continuous_map.homotopy_rel`.